listenbrainz_spark/stats/user/recording.py

```diff
@@ -13,64 +12,56 @@ def get_recordings(table):
     Returns:
         iterator (iter): an iterator over result:
                 {
-                    'user1' : [{
-                        'track_name': str,
-                        'recording_msid': str,
-                        'recording_mbid': str,
-                        'artist_name': str,
-                        'artist_msid': str,
-                        'artist_mbids': str,
-                        'release_name': str,
-                        'release_msid': str,
-                        'release_mbid': str,
-                        'listen_count': int
-                    }],
+                    'user1' : [
+                        {
+                            'recording_name': str,
+                            'recording_mbid': str,
+                            'artist_name': str,
+                            'artist_credit_id': int,
+                            'release_name': str,
+                            'release_mbid': str,
+                            'listen_count': int
+                        }
+                    ],
                     'user2' : [{...}],
                 }
     """
-    result = run_query("""
-              WITH intermediate_table as (
-                SELECT user_name
-                     , track_name
-                     , CASE
-                         WHEN recording_mbid IS NOT NULL AND recording_mbid != '' THEN NULL
-                         ELSE nullif(recording_msid, '')
-                       END as recording_msid
-                     , nullif(recording_mbid, '') as recording_mbid
-                     , artist_name
-                     , CASE
-                         WHEN cardinality(artist_mbids) > 0 THEN NULL
-                         ELSE nullif(artist_msid, '')
-                       END as artist_msid
-                     , artist_mbids
-                     , nullif(release_name, '') as release_name
-                     , CASE
-                         WHEN release_mbid IS NOT NULL AND release_mbid != '' THEN NULL
-                         ELSE nullif(release_msid, '')
-                       END as release_msid
-                     , nullif(release_mbid, '') as release_mbid
-                  FROM {}
-              )
-            SELECT *
+    result = run_query(f"""
```

Comment:
oh wow. I love how this PR simplifies things overall, but this is serious improvement!